dota_profanity_checker/game.py

```diff
@@ -0,0 +1,89 @@
+from pydantic import BaseModel, validator
+from typing import List
+
+from dota_profanity_checker.chat import Chat
+from dota_profanity_checker.player import Player
+from dota_profanity_checker.profanity import ProfanityChecker
+
+
+class Game(BaseModel):
+    """Class representing a single dota game."""
+    match_id: int
+    players: List[Player]
+    chat: List[Chat] = None
+    _chat_assigned: bool = False
+    _profanity_checker = ProfanityChecker()
+
+    class Config:
+        underscore_attrs_are_private = True
+        validate_assignment = True
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    @validator("chat")
+    def set_chat(cls, chat):
```

Comment:
Looking up how this `@validator` dsl is used - `set_chat` looks like it's a validation method that runs when the `chat` property is assigned. In a few languages I know, `set` and `get` are for [getter and setter methods](https://www.google.com/search?q=getter+and+setter+methods&source=lmns&bih=1297&biw=2560&rlz=1C1GCEA_enGB940GB940&hl=en-GB&sa=X&ved=2ahUKEwjznp-b4obzAhUR4BoKHWnnB0YQ_AUoAHoECAEQAA). Therefore, this was a bit confusing to read, as this is a _validation_ method that's being run when chat is set, called `set_chat`, that doesn't set the chat.

I'd suggest a method name that more clearly reflects its intent.

```suggestion
    def chat_present(cls, chat):
```

Reply:
It technically does both, it validates if chat is present and sets a default value (an empty list) if it isn't. I'll be honest, this @validator decorator is a quirk of the pydantic.BaseModel that I don't really understand. In a regular class this would be much simpler to do. I think I dove in a little quick with using pydantic! Have changed the name for now however.